src/streamlink_cli/main.py

```diff
@@ -138,19 +142,35 @@ def create_http_server(*_args, **_kwargs):
     return http
 
 
+def apply_format_vars(plugin, unformatted_str):
+    return LazyFormatter.format(
+        unformatted_str,
+        title=lambda: plugin.get_title() or DEFAULT_STREAM_METADATA["title"],
+        author=lambda: plugin.get_author() or DEFAULT_STREAM_METADATA["author"],
+        category=lambda: plugin.get_category() or DEFAULT_STREAM_METADATA["category"],
+        game=lambda: plugin.get_category() or DEFAULT_STREAM_METADATA["game"],
+        url=plugin.url
+    )
+
+
+def create_filename(plugin=None, unformatted_str=None):
+    if plugin and unformatted_str and unformatted_str != "-":
+        try:
+            return pathvalidate.sanitize_filename(
+                apply_format_vars(plugin, unformatted_str),
+                replacement_text="_"
+            )
+        except pathvalidate.ValidationError as err:
+            console.exit("Failed to sanitize output filename: {0}", err)
```

Comment:
> pathvalidate.sanitize_filename

According to the docs, this also replaces `/` with the replacement character, which means it's not possible anymore to choose different directories. This needs to be changed to `sanitize_filepath`:
https://github.com/thombashi/pathvalidate#sanitize-a-filename
https://github.com/thombashi/pathvalidate#sanitize-a-filepath

This does keep characters like `:` on non-Windows OSes, right?
https://github.com/thombashi/pathvalidate/blob/v2.4.1/pathvalidate/_filename.py#L85-L89
https://github.com/thombashi/pathvalidate/blob/v2.4.1/pathvalidate/_base.py#L14-L17

Reply:
> According to the docs, this also replaces / with the replacement character, which means it's not possible anymore to choose different directories. This needs to be changed to sanitize_filepath:

Ah, yes, you're absolutely right!

> This does keep characters like : on non-Windows OSes, right?

Not in universal mode.  If you think it's better to keep characters like : for an OS that can support them in filenames, then that be done by running pathvalidate in auto mode instead.  The side effect of that would be different filenames for the same substitution vars depending on the OS used.  I don't know if that matters or if consistent filenames are better?